mods/wool/init.lua

```diff
@@ -2,46 +2,68 @@
 
-local dyes = {
-	{"white",      "White",      "basecolor_white"},
-	{"grey",       "Grey",       "basecolor_grey"},
-	{"black",      "Black",      "basecolor_black"},
-	{"red",        "Red",        "basecolor_red"},
-	{"yellow",     "Yellow",     "basecolor_yellow"},
-	{"green",      "Green",      "basecolor_green"},
-	{"cyan",       "Cyan",       "basecolor_cyan"},
-	{"blue",       "Blue",       "basecolor_blue"},
-	{"magenta",    "Magenta",    "basecolor_magenta"},
-	{"orange",     "Orange",     "excolor_orange"},
-	{"violet",     "Violet",     "excolor_violet"},
-	{"brown",      "Brown",      "unicolor_dark_orange"},
-	{"pink",       "Pink",       "unicolor_light_red"},
-	{"dark_grey",  "Dark Grey",  "unicolor_darkgrey"},
-	{"dark_green", "Dark Green", "unicolor_dark_green"},
-}
-
-for i = 1, #dyes do
-	local name, desc, craft_color_group = unpack(dyes[i])
-
-	minetest.register_node("wool:" .. name, {
-		description = desc .. " Wool",
-		tiles = {"wool_" .. name .. ".png"},
+for key, value in pairs(dye.dyes) do
+	local color = key
+	local colorDisplayName = value.name
+	local htmlColor = value.html.."A0"
+	local craft_color_group = "color_"..color
+
+	minetest.register_node("wool:"..color, {
+		description = colorDisplayName.." Wool",
+		tiles = {"wool.png^[colorize:"..htmlColor},
 		is_ground_content = false,
-		groups = {snappy = 2, choppy = 2, oddly_breakable_by_hand = 3,
-				flammable = 3, wool = 1},
+		groups = {snappy = 2, choppy = 2, oddly_breakable_by_hand = 3, flammable = 3, wool = 1},
 		sounds = default.node_sound_defaults(),
 	})
 
-	minetest.register_craft{
+	minetest.register_craft({
 		type = "shapeless",
-		output = "wool:" .. name,
-		recipe = {"group:dye," .. craft_color_group, "group:wool"},
-	}
+		output = "wool:"..color,
+		recipe = {"group:dye,"..craft_color_group, "group:wool"},
+	})
 end
 
+--local dyes = {
+--	{"white",      "White"},
+--	{"grey",       "Grey"},
+--	{"black",      "Black"},
+--	{"red",        "Red"},
+--	{"yellow",     "Yellow"},
+--	{"green",      "Green"},
+--	{"cyan",       "Cyan"},
+--	{"blue",       "Blue"},
+--	{"magenta",    "Magenta"},
+--	{"orange",     "Orange"},
+--	{"violet",     "Violet"},
+--	{"brown",      "Brown"},
+--	{"pink",       "Pink"},
+--	{"dark_grey",  "Dark Grey"},
+--	{"dark_green", "Dark Green"},
+--}
+
+--for i = 1, #dyes do
+--	local name, desc = unpack(dyes[i])
+--	local craft_color_group = "color_"..name
+
+--	minetest.register_node("wool:" .. name, {
+--		description = desc .. " Wool",
+--		tiles = {"wool_" .. name .. ".png"},
+--		is_ground_content = false,
+--		groups = {snappy = 2, choppy = 2, oddly_breakable_by_hand = 3,
+--				flammable = 3, wool = 1},
+--		sounds = default.node_sound_defaults(),
+--	})
+
+--	minetest.register_craft{
+--		type = "shapeless",
+--		output = "wool:" .. name,
+--		recipe = {"group:dye," .. craft_color_group, "group:wool"},
+--	}
+--end
+
 
 -- legacy
 
 -- Backwards compatibility with jordach's 16-color wool mod
```

Comment:
I left the aliases commented out in case someone wanted them back in later, but I suppose that's what source control is for.